zerver/actions/streams.py

```diff
@@ -193,7 +193,9 @@ def do_reactivate_stream(
 
     # Update caches
     cache_set(display_recipient_cache_key(stream.recipient_id), new_name)
-    messages = Message.objects.filter(recipient_id=stream.recipient_id).only("id")
+    messages = Message.objects.filter(realm_id=realm.id, recipient_id=stream.recipient_id).only(
+        "id"
+    )
```

Comment:
Hmm, I worry a bit that this type of query in particular might turn out to be slower, depending what new indexes we've added; I'm not sure the query planner will know that `recipient_id` implies the value of `realm_id`. May be irrelevant with new indexes being added, but figured it worth mentioning.

(Not actually important in this case, since it's a rare code path)

Reply:
I see that you add the necessary indexes at the end of the branch. Posted some thoughts elsewhere but a potential theory would be to order this branch where the first commit adds the new indexes, then we migrate the code to take advantage of it with these new `realm_id` restrictions, and finally to remove the old indexes, possibly in a follow-up PR that we do once this is deployed.

Reply:
Sure, reordered to that order.